### PR TITLE
Implement ListHmacKeysReader and friends.

### PR DESCRIPTION
**Pull request by coryan**
This PR introduces the classes to iterate over the HMAC keys. Because
the iterators need a function on the `RawClient`, I made the changes to
support that too. A future PR will include the changes to
`storage::Client` and the corresponding test and examples.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2190)

### CODE REVIEW

File: google/cloud/storage/internal/hmac_key_requests.cc [hunk 1]
```diff
@@ -108,6 +108,14 @@ StatusOr<ListHmacKeysResponse> ListHmacKeysResponse::FromHttpResponse(
   return result;
 }
 
+std::ostream& operator<<(std::ostream& os, ListHmacKeysResponse const& r) {
+  os << "ListHmacKeysResponse={next_page_token=" << r.next_page_token
+     << ", items={";
+  std::copy(r.items.begin(), r.items.end(),
+            std::ostream_iterator<HmacKeyMetadata>(os, "\n  "));
```
Comment (devbww) [inline google/cloud/storage/internal/hmac_key_requests.cc]: Should there be newlines in a rendering?  At least without some highlighting "NOTE:"?

File: google/cloud/storage/list_hmac_keys_reader.h [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+#include "google/cloud/status_or.h"
+#include "google/cloud/storage/internal/hmac_key_requests.h"
+#include "google/cloud/storage/internal/range_from_pagination.h"
+#include "google/cloud/storage/internal/raw_client.h"
+#include <iterator>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
```
Comment (devbww) [inline google/cloud/storage/list_hmac_keys_reader.h]: #include "google/cloud/storage/version.h"

File: google/cloud/storage/internal/hmac_key_requests.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/hmac_key_requests.cc]: No opinion.  What do you prefer?

File: google/cloud/storage/list_hmac_keys_reader.h (hunk 2 above)
Comment (coryan) [reply google/cloud/storage/list_hmac_keys_reader.h]: Done.  But we have 3 other headers in this file that already *must* include `storage/version.h`.

File: google/cloud/storage/internal/hmac_key_requests.cc (hunk 1 above)
Comment (devbww) [reply google/cloud/storage/internal/hmac_key_requests.cc]: My first choice would be just comma separators, so we'd get something like ...

`ListHmacKeysResponse={next_page_token=blah, items={HmacKeyMetadata={...}, HmacKeyMetadata={...}}}`

That might get longish, but so be it unless/until we come up with some streaming format policy, like ...

```
ListHmacKeysResponse={
  next_page_token=blah, 
  items={
    HmacKeyMetadata={
      ...
    },
    HmacKeyMetadata={
      ...
    }
  }
}
```

File: google/cloud/storage/list_hmac_keys_reader.h (hunk 2 above)
Comment (devbww) [reply google/cloud/storage/list_hmac_keys_reader.h]: Granted, although it is nice if everything is independent.

File: google/cloud/storage/internal/hmac_key_requests.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/storage/internal/hmac_key_requests.cc]: Went with the commas.

**Pull request merged**